### PR TITLE
Reduces the size of the docker image for BikeSharing Web app

### PR DESCRIPTION
**Pull request by areai51**
Reduces the size of the docker image from ~1 GB to 137 MB by
- Using the node alpine
- Using Docker multistage build

### CODE REVIEW

**Comment by greenie-msft** [comment]
Hi @areai51, Did you open this PR by mistake?

**Comment by areai51** [comment]
No this PR was raised with the intent it would get merged into the main repo.
I noticed the docker images that gets built is quite large and this PR should drastically reduce it and eventually improve the speed of creating, storing and pulling the images from the container registry.

**Comment by greenie-msft** [comment]
@areai51 Got it! Thanks for improvement!

You might have made an extra commit in the PR.

In this commit "9ee2147b5a74ade4bac531fefdd9796d1fbcf3e3", it looks like you've committed extra changes that are specific to your cluster (host_suffix). If you can amend that commit, I can have the team review your PR.

**Comment by areai51** [comment]
Oh sorry!! just noticed it.. reverted that change, let me know if this looks ok

File: samples/BikeSharingApp/BikeSharingWeb/Dockerfile [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
+# STAGE-1: Build application
+FROM node:lts-alpine AS builder
 
 WORKDIR /app
 COPY package.json .
-RUN npm install
+RUN npm install --only=production
 COPY . .
 
+# STAGE-2: Copy built file into a new docker
```
Comment (DrEsteban) [inline samples/BikeSharingApp/BikeSharingWeb/Dockerfile]: I don't think the second build stage is necessary for this image.  Typically that convention is used to trim away any development-only artifacts from the image, but in this case it looks like they are all being copied over anyway.  Did you have another goal in mind by structuring it this way?

Comment (areai51) [reply samples/BikeSharingApp/BikeSharingWeb/Dockerfile]: true this is sort of just a foot in the door to put in a multi-stage step. In the next PR I'd like to  copy only the `./app/.next` folder but that would need a bit more work and I was hoping to look into it after this goes in.

Comment (DrEsteban) [reply samples/BikeSharingApp/BikeSharingWeb/Dockerfile]: Would you mind removing the second build stage for the purposes of this PR?  (And potentially following up with a fully-implemented "two stage build" PR?)  Sorry to nit-pick!

We have a desire for our `master` branch to guide users into a recommended set of practices, and we fear the current implementation may confuse users.

Comment (areai51) [reply samples/BikeSharingApp/BikeSharingWeb/Dockerfile]: Apologies for the delay, removed the second build stage.

Comment (DrEsteban) [reply samples/BikeSharingApp/BikeSharingWeb/Dockerfile]: Thanks!

**Pull request merged**